src/mpid/ch4/src/ch4r_recvq.c

```diff
@@ -35,5 +37,16 @@ int MPIDIG_recvq_init(void)
     MPIR_T_PVAR_TIMER_REGISTER_STATIC(RECVQ, MPI_DOUBLE, time_matching_unexpectedq, MPI_T_VERBOSITY_USER_DETAIL, MPI_T_BIND_NO_OBJECT, (MPIR_T_PVAR_FLAG_READONLY | MPIR_T_PVAR_FLAG_CONTINUOUS), "CH4",        /* category name */
                                       "total time spent on search passes on the unexpected receive queue");
 
+    int source_index;
+    MPIR_T_register_source("RECVQ", "active message receive queue", MPI_T_SOURCE_ORDERED, NULL, -1,
+                           -1, &source_index);
+
+    MPI_Datatype array_of_datatypes[1] = { MPI_INT };
+    MPI_Aint array_of_displacements[1] = { 0 };
+    MPIR_T_register_event(source_index, "unexp_message", MPI_T_VERBOSITY_USER_BASIC,
+                          array_of_datatypes, array_of_displacements, 1,
+                          "message added to unexpected queue", MPI_T_BIND_NO_OBJECT, "CH4",
+                          &unexp_message_index);
+
```

Comment:
Nice! I can use this! But to be useful, one would need to have dequeuing events as well or they wouldn't be able to monitor queue lengths -- the actual performance factor.

Reply:
A companion dequeue event makes sense and could be easily added. Note that we do have PVARs (i.e. not events) that monitor queue lengths.

Reply:
Added a dequeue event in latest push.